doc/extdev/index.rst

```diff
@@ -189,6 +189,123 @@ as metadata of the extension.  Metadata keys currently recognized are:
+
+Finally, in your main ``setup`` function, add your custom setup function
+as a callback for :event:`html-page-context`.
+
+.. code:: python
+
+   # The parent function
+   def setup_my_func(app, pagename, templatename, context, doctree):
+
+      # Define the template function
+      def my_func(mystring):
+          return "Your string is %s" % mystring
+
+      # Add it to the page's context
+      context['my_func'] = my_func
+
+   # Your extension's setup function
+   def setup(app):
+      app.connect("html-page-context", setup_my_func)
+
+Now, you will have access to this function in jinja like so:
+
+.. code:: jinja
+
+   <div>
+   {{ my_func("some string") }}
+   </div>
+
+Make an extension depend on another extension
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Sometimes your extension depends on the functionality of another
+Sphinx extension. Most Sphinx extensions are activated in a
+site's :file:`conf.py` file, but this is not available to you as an extension
+developer.
+
+.. module:: sphinx.application
+
+To ensure that another extension is activated as a part of your own extension,
+use the :meth:`Sphinx.setup_extension` method. This will
+activate another extension at run-time, ensuring that you have access to its
+functionality.
+
+For example, the following code activates the "recommonmark" extension:
+
+.. code:: python
+
+   def setup(app):
+      app.setup_extension("recommonmark")
+
+.. note::
+
+   Since your extension will depend on another, make sure to include it as a part
+   of your extension's installation requirements.
+
+Inject javsacript based on user configuration
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If your extension makes use of JavaScript, it can be useful to allow users
+to control its behavior using their Sphinx configuration. However, this can
+be difficult to do if your JavaScript comes in the form of a static library
+(which will not be built with Jinja).
+
+To inject variables into the JavaScript space based on user configuration,
+use the :meth:`Sphinx.add_js_file` method. Normally, this method is used to
+insert a new JavaScript file into your site. However, if you do *not* pass
+a file path, but instead pass a string to the "body" argument, then this
+text will be inserted as JavaScript into your site's head.
+
+For example, the following code will read in a user-configured value and then
+insert this value as a JavaScript variable, which your extension's JavaScript
+code may use:
+
+.. code:: python
+
+   # This function reads in a variable and inserts it into JavaScript
+   def add_js_variable(app):
+      # This is a configuration that you've specified for users in `conf.py`
+      js_variable = app.config['my_javascript_variable']
+      js_text = "var my_variable = '%s';" % js_variable
+      app.add_js_file(None, body=js_text)
+
+   # We connect this function to the step after the builder is initialized
+   def setup(app):
+      app.connect('builder-inited', add_js_variable)
```

Comment:
How about adding `app.add_config_value('my_javascript_variable')`? It makes this example copy-pastable!